File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/Python/s573993837_commented.py
class InputReader:
    def __init__(self, stream):
        self.reader = BufferedReader(InputStreamReader(stream), 32768)
        self.tokenizer = None

    def next(self):
        while self.tokenizer is None or not self.tokenizer.hasMoreTokens():
            try:
                self.tokenizer = StringTokenizer(self.reader.readLine())
            except IOException as e:
                raise RuntimeException(e)
        return self.tokenizer.nextToken()

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return int(self.next())

    def nextDouble(self):
        return float(self.next())


def solve(sc, pw):
    t = 1
    while t > 0:
        n = sc.nextLong()
        m = sc.nextLong()
        if n == 0:
            pw.write(str(0) + '\n')
            continue
        if n == m:
            pw.write(str(0) + '\n')
            continue
        if n > m:
            n = m + n % m
            n -= m
            pw.write(str(min(m - n, n)) + '\n')
            continue
        pw.write(str(min(m - n, n)) + '\n')


def main():
    input_stream = sys.stdin
    output_stream = sys.stdout
    in_reader = InputReader(input_stream)
    out_writer = output_stream
    solve(in_reader, out_writer)
    out_writer.flush()


if __name__ == "__main__":
    main()  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/C++/s573993837_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <cmath>

using namespace std;

class InputReader {
public:
    string next() {
        while (!tokenizer.empty()) {
            tokenizer.pop_back();
        }
        string line;
        getline(cin, line);
        stringstream ss(line);
        string token;
        while (ss >> token) {
            tokenizer.push_back(token);
        }
        if (tokenizer.empty()) {
            return "";
        }
        string result = tokenizer.front();
        tokenizer.erase(tokenizer.begin());
        return result;
    }

    int nextInt() {
        return stoi(next());
    }

    long long nextLong() {
        return stoll(next());
    }

    double nextDouble() {
        return stod(next());
    }

private:
    vector<string> tokenizer;
};

void solve(InputReader &in, ostream &out) {
    int t = 1;
    while (t--) {
        long long n = in.nextLong();
        long long m = in.nextLong();
        if (n == 0) {
            out << 0 << endl;
            continue;
        }
        if (n == m) {
            out << 0 << endl;
            continue;
        }
        if (n > m) {
            n = m + n % m;
            n -= m;
            out << min(m - n, n) << endl;
            continue;
        }
        out << min(m - n, n) << endl;
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    InputReader in(cin);
    ostream &out = cout;

    solve(in, out);

    return 0;
}

// End of Code
